src/combine.jl

```diff
@@ -2,48 +2,55 @@ import Base: merge, hcat, vcat, map
 
 ###### merge ####################
 
-function _merge_outer(::Type{IndexType}, ta1::TimeArray{T, N, D}, ta2::TimeArray{T, M, D}, padvalue, meta) where {IndexType, T, N, M, D}
-    timestamps, new_idx1, new_idx2 = sorted_unique_merge(IndexType, ta1.timestamp, ta2.timestamp)
-    vals = fill(convert(T, padvalue), (length(timestamps), length(ta1.colnames) + length(ta2.colnames)))
-    insertbyidx!(vals, ta1.values, new_idx1)
-    insertbyidx!(vals, ta2.values, new_idx2, size(ta1.values, 2))
-    TimeArray(timestamps, vals, [ta1.colnames; ta2.colnames], meta; unchecked = true)
+function _merge_outer(::Type{IndexType}, ta1::TimeArray{T,N,D}, ta2::TimeArray{T,M,D}, padvalue, meta) where {IndexType,T,N,M,D}
+    timestamps, new_idx1, new_idx2 = sorted_unique_merge(IndexType, timestamp(ta1), timestamp(ta2))
+    vals = fill(convert(T, padvalue), (length(timestamps), length(colnames(ta1)) + length(colnames(ta2))))
+    insertbyidx!(vals, values(ta1), new_idx1)
+    insertbyidx!(vals, values(ta2), new_idx2, size(values(ta1), 2))
+    TimeArray(timestamps, vals, [colnames(ta1); colnames(ta2)], meta; unchecked = true)
 end
 
-function merge(ta1::TimeArray{T, N, D}, ta2::TimeArray{T, M, D}, method::Symbol=:inner;
-               colnames::Vector=[], meta::Any=nothing, padvalue=NaN) where {T, N, M, D}
+function merge(ta1::TimeArray{T,N,D}, ta2::TimeArray{T,M,D}, method::Symbol = :inner;
+               colnames::Vector = Symbol[], meta = nothing,
+               padvalue=NaN) where {T,N,M,D}
 
-    if ta1.meta == ta2.meta && meta isa Nothing
-        meta = ta1.meta
-    elseif typeof(ta1.meta) <: AbstractString && typeof(ta2.meta) <: AbstractString && meta isa Nothing
-        meta = string(ta1.meta, "_", ta2.meta)
+    if colnames isa Vector{<:AbstractString}
+        @warn "`merge(...; colname::Vector{<:AbstractString})` is deprecated, " *
+              "use `merge(...; colnames=Symbol.(colnames))` instead."
+        colnames = Symbol.(colnames)
+    end
+
+    meta = if _meta(ta1) == _meta(ta2) && meta isa Nothing
```

Comment:
`meta === nothing` is typically preferred to `meta isa Nothing`. In particular, `===` makes the compiler happy. (Though `isa` is still better than `==` in this case.)

Reply:
ha, okay 5d1dcc7